### ISSUE TITLE
Release 6.0.0-rc.1

### ISSUE THREAD

**Issue by TheSnoozer** [git-commit-id/git-commit-id-plugin-core#12]
This is the place holder to release the new 6.0.0-rc.1 version.
Previous releases can be found under https://repo1.maven.org/maven2/pl/project13/maven/git-commit-id-plugin-core/

Note that the current open tickets should be addressed, before performing a release.

# General scope of the new release:
Fixed some minor things, added some minor features...(full list https://github.com/git-commit-id/git-commit-id-plugin-core//issues?q=milestone%3A5.0.0).

# Release-Guide
see http://central.sonatype.org/pages/ossrh-guide.html

- [x] update dependencies `mvn versions:display-dependency-updates`
- [x] wait for github actions to pass
- [x] `mvn release:prepare`
[INFO] Checking dependencies and plugins for snapshots ...
What is the release version for "Git Commit Id Plugin Core"? (io.github.git-commit-id:git-commit-id-plugin-core) **6.0.0-rc.1**: : [ENTER]
What is SCM release tag or label for "Git Commit Id Plugin Core"? (io.github.git-commit-id:git-commit-id-plugin-core) git-commit-id-plugin-core-6.0.0-rc.1: : **v6.0.0-rc.1** [ENTER]
What is the new development version for "Git Commit Id Plugin Core"? (io.github.git-commit-id:git-commit-id-plugin-core) **6.0.0-SNAPSHOT**: : [ENTER]
- [x] wait for github actions to pass
- [x] `mvn release:perform`
  - (or `mvn clean source:jar javadoc:jar deploy -Pgpg` from the git tag)
  - (or `mvn release:perform -Dresume=false`)
  - Note: If the uploading of the artifacts fails, ensure that a [`settings.xml`](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/master/.buildscript/settings.xml) exists under the local `.m2`-Folder
- [x] then go to https://s01.oss.sonatype.org/ log in there and go to the staging repositories, there will be the plugin, you have to first close and then release it if validation passed.
- [x] verify plugin is available on (might take some time) https://repo1.maven.org/maven2/io/github/git-commit-id/git-commit-id-plugin-core/
- [x] under [Milestones](https://github.com/git-commit-id/git-commit-id-plugin-core/milestones) close old milestone
- [x] under [Milestones](https://github.com/git-commit-id/git-commit-id-plugin-core/milestones) create new milestone for new version
- [ ] under [Releases](https://github.com/git-commit-id/git-commit-id-plugin-core/releases) publish Release-Notes
- [ ] update readme to point to the latest version available

**Comment by TheSnoozer** [comment]
Ugh. A `mvn release:perform` fails since the release plugin performs a full-checkout:
```
[INFO] 2/3 perform:checkout-project-from-scm
[INFO] Checking out the project to perform the release ...
[INFO] Executing: /bin/sh -c cd '/git-commit-id/git-commit-id-plugin-core/target' && 'git' 'clone' '--depth' '1' '--branch' 'v6.0.0-rc.1' 'git@github.com:git-commit-id/git-commit-id-plugin-core.git' 'checkout'
[INFO] Working directory: git-commit-id/git-commit-id-plugin-core/target
[INFO] Executing: /bin/sh -c cd '/tmp' && 'git' 'ls-remote' 'git@github.com:git-commit-id/git-commit-id-plugin-core.git'
[INFO] Working directory: /tmp
[INFO] Executing: /bin/sh -c cd '/git-commit-id/git-commit-id-plugin-core/target/checkout' && 'git' 'fetch' 'git@github.com:git-commit-id/git-commit-id-plugin-core.git'
[INFO] Working directory: /git-commit-id/git-commit-id-plugin-core/target/checkout
[INFO] Executing: /bin/sh -c cd '/git-commit-id/git-commit-id-plugin-core/target/checkout' && 'git' 'checkout' 'v6.0.0-rc.1'
[INFO] Working directory: /git-commit-id/git-commit-id-plugin-core/target/checkout
[INFO] Executing: /bin/sh -c cd '/git-commit-id/git-commit-id-plugin-core/target/checkout' && 'git' 'ls-files'
[INFO] Working directory: /git-commit-id/git-commit-id-plugin-core/target/checkout
```
However that does not consider submodules and makes the test fail (remember test resources are organized as submodules).

Investigate solutions like
- the localCheckout-option (https://maven.apache.org/maven-release/maven-release-plugin/perform-mojo.html#localCheckout)
- https://sumit-jha.medium.com/how-to-make-maven-release-plugin-work-with-git-submodules-917825a253f2

I guess this release can be redirected to the dumpster.

**Comment by TheSnoozer** [comment]
See https://issues.apache.org/jira/browse/SCM-530 (a 13 year old request we need to workaround now...great stuff).

**Comment by TheSnoozer** [comment]
I now have performed the release again, this time with the workaround of cloning the submodules.

Released as v6.0.0-rc.2.
It is idential to v6.0.0-rc.1 (besides the workaround for cloning submodules).

Perhaps as follow-up:
- Have a validation that fails if ` ~/.m2/settings.xml` is not present. Otherwise one get's cryptic errors while uploading the artifacts to sonatype.
- There seems to be a plugin that autoCloses things:
```xml
<distributionManagement>
  <snapshotRepository>
    <id>ossrh</id>
    <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
  </snapshotRepository>
</distributionManagement>
<build>
  <plugins>
    <plugin>
      <groupId>org.sonatype.plugins</groupId>
      <artifactId>nexus-staging-maven-plugin</artifactId>
      <version>1.6.7</version>
      <extensions>true</extensions>
      <configuration>
        <serverId>ossrh</serverId>
        <nexusUrl>https://s01.oss.sonatype.org/</nexusUrl>
        <autoReleaseAfterClose>true</autoReleaseAfterClose>
      </configuration>
    </plugin>
    ...
  </plugins>
</build>
```
since I usually just close the things anyway it may be worth adding that too.

**Comment by TheSnoozer** [comment]
Release now available under
https://repo1.maven.org/maven2/io/github/git-commit-id/git-commit-id-plugin-core/6.0.0-rc.2/
